rotkehlchen/db/dbhandler.py

```diff
@@ -2358,6 +2358,41 @@ def delete_bitcoin_xpub(self, xpub_data: XpubData) -> None:
         self.conn.commit()
         self.update_last_write()
 
+    def edit_bitcoin_xpub(self, xpub_data: XpubData) -> None:
+        """Edit the xpub tags and label
+
+        May raise:
+        - InputError if the xpub data already exist
+        """
+        cursor = self.conn.cursor()
+        key = xpub_data.xpub.xpub + xpub_data.serialize_derivation_path_for_db()  # type: ignore
+        try:
+            # Delete the tag mappings for the xpub itself (type ignore is for xpub is not None
+            cursor.execute('DELETE FROM tag_mappings WHERE object_reference=?', (key,))
+
+            insert_tag_mappings(
+                # if we got tags add them to the xpub
+                cursor=cursor,
+                data=[xpub_data],
+                object_reference_keys=['xpub.xpub', 'derivation_path'],
+            )
+
+            cursor.execute(
+                'UPDATE xpubs SET label=? WHERE xpub=? AND derivation_path=?',
+                (
+                    xpub_data.label,
+                    xpub_data.xpub.xpub,
+                    xpub_data.serialize_derivation_path_for_db(),
+                ),
+            )
+        except sqlcipher.IntegrityError:  # pylint: disable=no-member
+            raise InputError(
```

Comment:
The error message here still refers to adding an xpub. I think the only way an integrity error can be raised here is if you try to edit and add an already existing tag to the same xpub.

Can you test this? By writing another test?

Perhaps a solution to this would also e to edit the `insert_tag_mappings()` function and change the statement from `INSERT` to `INSERT OR IGNORE`

Reply:
> The error message here still refers to adding an xpub. I think the only way an integrity error can be raised here is if you try to edit and add an already existing tag to the same xpub.
That cannot happen though since all mapping for the xpub are removed first right?

> Can you test this? By writing another test?

I will move the insert_tag_mappings outside the try catch and let any exception bubble up. Since the caller already checks for the TagConstraint exception

> Perhaps a solution to this would also e to edit the `insert_tag_mappings()` function and change the statement from `INSERT` to `INSERT OR IGNORE`

I don't think that would make sense. We already assume that you have created the tags before adding them anywhere in the app.

Reply:
> I don't think that would make sense. We already assume that you have created the tags before adding them anywhere in the app.

This is not about the tags themselves. But this is for the tag **mappings**.

https://github.com/rotki/rotki/blob/9326be49446ea889302439bd73304eb9fc7006b6/rotkehlchen/db/utils.py#L116-L139

If you try to map a tag to someting that it's already mapped to, so if the mapping already exists then it will probably raise an integrity error. This is what I asked you to test.